src/aspire/abinitio/commonline_d2.py

```diff
@@ -0,0 +1,1888 @@
+                    )
+                    true_scls_corrs = corrs[scl_idx_list]
+                    scls_cand_idx = self.scl_idx_lists[1, eq_idx, j]
+                    eq_measures_j = eq_measures[scls_cand_idx]
+                    measures_agg = np.outer(true_scls_corrs, eq_measures_j)
+                    k = self.non_tv_eq_idx[eq_idx]
+                    corrs_out[i, k * n_inplane + j] = np.max(measures_agg)
+
+        self.scls_scores = corrs_out
+
+    def _all_eq_measures(self, corrs):
+        """
+        Compute a measure of how much an image from data is close to an equator.
+
+        :param corrs: Correlation table of shape (n_theta // 2, n_theta).
+
+        :return: (n_theta // 2) likelihood scores.
+        """
+        # First compute the eq measure (corrs(scl-k,scl+k) for k=1:90)
+        # An equator image of a D2 molecule has the following property: If t_i is
+        # the angle of one of the rays of the self common line then all the pairs of
+        # rays of the form (t_i-k,t_i+k) for k=1:90 are identical. For each t_i we
+        # average over correlations between the lines (t_i-k,t_i+k) for k=1:90
+        # to measure the likelihood that the image is an equator and the ray (line)
+        # with angle t_i is a self common line.
+        # (This first loop can be done once outside this function and then pass
+        # idx as an argument).
+        L = self.n_theta
+        idx = np.zeros((L // 2, L // 4, 2))
+        idx_1 = np.mod(
+            np.vstack((-np.arange(1, L // 4 + 1), np.arange(1, L // 4 + 1))), L
+        )
+        idx[0, :, :] = idx_1.T
+        for k in range(1, L // 2):
+            idx[k, :, :] = np.mod(idx_1.T + k, L)
+        idx = np.mod(idx, L)
+
+        # Convert to Fourier ray indices.
+        idx_1 = idx[:, :, 0].flatten()
+        idx_2 = idx[:, :, 1].flatten()
+
+        # Make all Ri coordinates < 180 and compute linear indices for corrrelations
+        is_geq_than_pi = idx_1 >= L // 2
+        idx_1[is_geq_than_pi] = idx_1[is_geq_than_pi] - (L // 2)
+        idx_2[is_geq_than_pi] = (idx_2[is_geq_than_pi] + (L // 2)) % L
+
+        # Compute correlations.
+        eq_corrs = corrs[idx_1.astype(int), idx_2.astype(int)]
+        eq_corrs = eq_corrs.reshape(L // 2, L // 4)
+        corrs_mean = np.mean(eq_corrs, axis=1)
+
+        # Now compute correlations for normals to scls.
+        # An eqautor image of a D2 molecule has the additional following property:
+        # The normal line to a self common line in 2D Fourier plane is real valued
+        # and both of its rays have identical values. We use the correlation
+        # between one Fourier ray of the normal to a self common line candidate t_i
+        # with its anti-podal as an additional way to measure if the image is an
+        # equator and t_i+0.5*pi is the normal to its self common line.
+        r = 2  # Search radius within 2 adjacent rays of normal ray.
```

Comment:
Not really sure. It was not configurable in matlab. In matlab this would have always represented 2 degrees since they were only working with `n_theta=360`. In our case it's the two adjacent rays. I wasn't really sure how to handle this at the time.

I could change it to `np.ceil(2 * n_theta / 360)` and that might be more in the spirit of the original intention.

Reply:
Yeah, probably `np.ceil(2 * n_theta / 360)` should be implemented if the author wanted rays...

I have noticed a lot of the matlab code has algo config vars hard coded.  I'll leave it up to you and Joakim to decide if it should be a parameter in the next review round.

Reply:
Just to be clear, if we leave it as `r=2` we are picking the two adjacent rays no matter what value for `n_theta` we are using. If we set it to `r = np.ceil(2 * n_theta / 360)` we are picking however many adjacent rays account for 2 degrees. So if `n_theta = 180`, we have `r=1` since each `pf` ray is separated by 2 degrees.

Reply:
err my bad; I just meant do what you feel the author wanted, but adapted for different `n_theta`.

Reply:
Changed this to `np.ceil(2 * n_theta / 360)` for now.